lib/AST/DiagnosticEngine.cpp

```diff
@@ -32,6 +32,7 @@
 #include "swift/Config.h"
 #include "swift/Localization/LocalizationFormat.h"
 #include "swift/Parse/Lexer.h" // bad dependency
+#include "../Serialization/ModuleFile.h"
```

Comment:
I think it would be good to avoid this:
1. It's a private header in `Serialization`. If we wanted to use it we really should move it out into the top level `include`.
2. This introduces a circular dependency - `Serialization` depends on `ClangImporter` depends on `AST`. And `AST` depends on `Serialization` now.

Could we just add the new `diagnose` in `Serialization` and have it do the `ModuleFile` -> `SourceLoc` mapping instead? That would also avoid all the `{}` -> `SourceLoc()` changes.

Reply:
I added a `ModuleFile::getSourceLoc()` instead, it cleans up the PR nicely.